doc/howto/dev/build_en.md

```diff
@@ -0,0 +1,100 @@
+# Build PaddlePaddle from Source Code and Run Unit Test
+
+## What Developers Need
+
+To contribute to PaddlePaddle, you need
+
+1. A computer -- Linux, BSD, Windows, MacOS, and
+1. Docker.
+
+Nothing else.  Not even Python and GCC, because you can install all build tools into a Docker image.
```

Comment:
do we need to add bit more explanation: think the docker container as a standalone factory, it takes paddlepaddle source code and produce paddlepaddle binary. Developer plays the role of a regular consumer who doesn't need to know any detail of the factory, and just use/test the product of the factory.

Reply:
I am afraid that it is intractable to explain Docker clearly enough in this document, so I have a hyperlink later pointing to an official Docker document.